superbit_lensing/shear_profiles/annular_jmac.py

```diff
@@ -84,106 +111,156 @@ def get_r_gtan(self,write=False):
+    def compute_profile(self, outfile, overwrite=False):
 
         """
         workhorse; this will contain the actual azimuthal binning
         """
+
         minrad = self.annular_info['rad_args'][0]
         maxrad = self.annular_info['rad_args'][1]
         num_bins = self.annular_info['nbins']
 
-        number,bins = np.histogram(self.r,bins=num_bins,range=(minrad,maxrad))
+        bins = np.linspace(minrad, maxrad, num_bins)
 
+        counts, bins = np.histogram(self.r, bins=bins)
 
         print("##\n r       n        gtan      err_gtan     gcross     err_gcross")
 
+        N = len(bins) - 1
+        midpoint_r = np.zeros(N)
+        gtan_mean = np.zeros(N)
+        gtan_err = np.zeros(N)
+        gcross_mean = np.zeros(N)
+        gcross_err = np.zeros(N)
+
+        # for i in range(len(bins)-1):
+        i = 0
+        for b1, b2 in zip(bins[:-1], bins[1:]):
+            # annulus = (self.r>=bins[i]) & (self.r<bins[i+1])
+            annulus = (self.r >= b1) & (self.r < b2)
+            n = counts[i]
+            midpoint_r[i] = np.mean([b1, b2])
+            gtan_mean[i] = np.mean(self.gtan[annulus])
+            gcross_mean[i] = np.mean(self.gcross[annulus])
+            gtan_err[i] = np.std(self.gtan[annulus]) / np.sqrt(n)
+            gcross_err[i] = np.std(self.gcross[annulus]) / np.sqrt(n)
+
+            print("%.3f   %d   %f   %f   %f   %f" %\
+                  (midpoint_r[i], n, gtan_mean[i], gtan_err[i], gcross_mean[i], gcross_err[i])
+                  )
+
+            i += 1
+
+        table = Table()
+        table.add_columns(
+            [counts, midpoint_r, gtan_mean, gcross_mean, gtan_err, gcross_err],
+            names=[
+                'counts', 'midpoint_r', 'gtan_mean', 'gcross_mean', 'gtan_err', 'gcross_err'
+                ],
+            )
+
+        table.write(outfile, format='fits', overwrite=overwrite)
+
+        return
 
-        for i in range(len(bins)-1):
-            annulus = (self.r>=bins[i]) & (self.r<bins[i+1])
-            midpoint_r = np.mean([bins[i],bins[i+1]])
-            n = number[i]
-            gtan_mean = np.mean(self.gtan[annulus])
-            gcross_mean = np.mean(self.gcross[annulus])
-            gtan_err = np.std(self.gtan[annulus])/np.sqrt(n)
-            gcross_err = np.std(self.gcross[annulus])/np.sqrt(n)
+    def plot_profile(self, cat_file, truth_file, plot_file):
 
-            print("%.3f   %d   %f   %f   %f   %f" % (midpoint_r,n,gtan_mean,gtan_err,gcross_mean,gcross_err))
+        # plotter = ShearProfilePlotter(cat_file, truth_file)
+        # plotter.plot(plot_file)
 
         return
 
+    def run(self, outfile, plotfile, overwrite=False):
 
-    def run(self,write=False):
+        outdir = os.path.dirname(outfile)
 
         self.open_table(self.cat_info)
-        self.get_r_gtan(write=write)
-        self.do_annular()
+        self.transform_shears(outdir, overwrite=overwrite)
+        self.compute_profile(outfile, overwrite=overwrite)
```

Comment:
These names are better than `get_r_gtan()` and `do_annular()`.

Reply:
I did low-key like `do_annular()` tho